### PR TITLE
Add note about when to use HGlobal allocator APIs.

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
Fixes https://github.com/dotnet/dotnet-api-docs/issues/10398

### CODE REVIEW

File: xml/System.Runtime.InteropServices/Marshal.xml [hunk 1]
```diff
@@ -303,6 +303,9 @@
           <format type="text/markdown"><![CDATA[
 
 ## Remarks
+> [!IMPORTANT]
+>  This native memory allocator is a legacy API that should be used exclusively when called for by specific Win32 APIs on the Windows platform. On non-Windows platforms, the <xref:System.Runtime.InteropServices.NativeMemory> class should be used to allocate native memory.
```
Comment (gewarren) [inline xml/System.Runtime.InteropServices/Marshal.xml:307]:
```suggestion
> This native memory allocator is a legacy API that should be used exclusively when called for by specific Win32 APIs on the Windows platform. On non-Windows platforms, use the <xref:System.Runtime.InteropServices.NativeMemory> class to allocate native memory.
```

**Review by gewarren — APPROVED** [review]

File: xml/System.Runtime.InteropServices/Marshal.xml [hunk 2]
```diff
@@ -303,6 +303,9 @@
           <format type="text/markdown"><![CDATA[
 
 ## Remarks
+> [!IMPORTANT]
+> This native memory allocator is a legacy API that should be used exclusively when called for by specific Win32 APIs on the Windows platform. On non-Windows platforms, use the <xref:System.Runtime.InteropServices.NativeMemory> class to allocate native memory.
```
Comment (jkotas) [inline xml/System.Runtime.InteropServices/Marshal.xml:307]: This is incomplete guidance. What should people do on Windows platforms or when they target TFM that does not have `S.R.I.NativeMemory`?

Comment (AaronRobinsonMSFT) [reply xml/System.Runtime.InteropServices/Marshal.xml:307]: I appreciate what you're getting at, but I don't think the doc will matter in practice all that much for those cases (TFM < .NET 6). Perhaps all that is needed in this case then is to say:

```
This native memory allocator is a legacy API that should be used exclusively when called for by specific Win32 APIs on the Windows platform or when targeting .NET versions earlier than .NET 6.0. On non-Windows platforms, use the <xref:System.Runtime.InteropServices.NativeMemory> class to allocate native memory.
```

**Comment by AaronRobinsonMSFT** [comment]
@gewarren Can you take one more look at my updated phrasing? Thanks.

**Review by gewarren — APPROVED** [review]
LGTM

**Pull request merged**